R/DIFFUSE.R

```diff
@@ -0,0 +1,266 @@
+#' 'heat' (gene score) put on node j. For all E(i,j) where there exists a directed edge from
+#' j to i, edge j -> i is equal to the value of E(i,j).
+#'
+#' @export
+
+DIFFUSE <- function(AGG = NULL, algorithm = "Leis", param = list(getOption("rete.beta")),
+                    silent = FALSE, noLog = FALSE) {
+        startTime <- Sys.time()
+
+        consoleVect <- paste("Diffuse started at", Sys.time(), sep = "")
+        if (!silent) {
+            print(consoleVect) #print beginning of function to console
+        }
+#========= Checking Arguments =============================
+        consoleVect <- "Checking Arguments"
+        if (!silent) {
+            print(consoleVect) #print beginning of function to console
+        }
+
+        #Check if AGG is an igraph object, and check if it's directed
+        if (!(igraph::is.igraph(AGG) && igraph::is.directed(AGG))) {
+            stop("Error: AGG must be an iGraph object with directed edges")
+        }
+
+
+        #Check other argmuments
+        check1 <- .checkArgs(algorithm,character(length = 1),checkSize = TRUE)
+        check2 <- .checkArgs(silent, logical(length = 1), checkSize = TRUE)
+        check3 <- .checkArgs(noLog, logical(length = 1), checkSize = TRUE)
+        check4 <- .checkArgs(param, list())
+
+        if (length(c(check1,check2,check3,check4)) > 0 ) {
+            stop("Error: incorrect argument(s) for algorithm, silent, noLog, and/or param.
+                 See documentation for DIFFUSE function")
+        }
+
+        #Extract AGG metadata for writing to log
+        metaAGG <- attr(AGG,"meta")
+           #Assign AGG vertices and edges to data frames
+           AGGverts<- igraph::as_data_frame(AGG, "vertices")
+           AGGedges<- igraph::as_data_frame(AGG, "edges")
+
+           sizeEdges <- nrow(AGGedges) #Number of edges, used for indexing purposes
+           sizeVerts <- nrow(AGGverts) #Number of vertices, used for indexing purposes
+
+    if (algorithm == "Leis") {
+
+        consoleVect<- "Method Selected: Leis"
+        if (!silent) {
+            print(consoleVect)
+        }
+
+        #check beta
+        if ( length(.checkArgs(param[[1]], numeric(length = 1))) > 1) {
+            stop("Leis Method selected. Beta must be class numeric of type
+                 double")
+        }
+
+        #creating transition matrix W
+        zeroVect <- numeric(length(sizeVerts)^2) #Make a vector of zeros from which to make W
+        W<-matrix(data = zeroVect, nrow = sizeVerts , ncol = sizeVerts,
+                  dimnames = list(AGGverts$name, AGGverts$name))
+
+        vertDegree <- numeric(length = sizeVerts) #Will store degree of each vertex
+        names(vertDegree) <- AGGverts$name
+
+        consoleVect<-"Calculating degree for vertices"
+        if (!silent) {
+            print(consoleVect)
+        }
+        #calculating degree via following procedure
+
+        HGNCsymb<-AGGverts$name #get a vector of AGG vertex names
```

Comment:
You use ```AGGverts$name``` four times before you assign it here. Either assign it right away, or keep on using ```AGGverts$name```.